src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacFreeTextValidator.java

```diff
@@ -61,8 +65,20 @@ private static String convertToPMatchWildcardNotation(final String ruleValue) {
         return ouSyntaxRuleValue;
     }
 
-    private static String extractAnswerValue(Choice answer, FreeTextRule rule) {
-        return rule.isCaseInsensitive() ? answer.getValue().toLowerCase() : answer.getValue();
+    private String removeNonAlphanumericChars(final String answer, final String rule) {
+        String strippedAnswer = answer;
+        List<String> nonAlphanumericCharsNotInRule = NON_ALPHANUMERIC_CHARS.codePoints()
+                .mapToObj(nonAlphaCodePoint -> String.valueOf((char) nonAlphaCodePoint))
+                .filter(nonAlphaString -> !rule.contains(nonAlphaString))
+                .collect(Collectors.toList());
+        for (String nonAlphanumericChar : nonAlphanumericCharsNotInRule) {
```

Comment:
I think this could just loop over all characters in the `NON_ALPHANUMERIC_CHARS` and only replace them if they do not feature in the rule? It would avoid looping twice, and not require anything complicated like streams?

Reply:
Good idea!